www/views.py

```diff
@@ -458,3 +458,45 @@ def createuser(request):
     else:
         form = CreateUserForm()
     return render(request, "www/createuser.html", {"userform": form})
+
+@login_required
+@staff_member_required
+def deleteuser(request, id, delete):
+    user = CustomUser.objects.get(id=id)
+    print (user.state)
+    if not (user.state == CustomUser.MEMBER or user.state == CustomUser.RESIGN_DELETE or user.state == CustomUser.RESIGN_SAVE_ACCOUNT):
+        messages.error(
+            request,
+                    _(
+                        "User not member or requesting delete"
+                    ),
+                )
+        return userdetails(request, id)
+
+    custominvoices = CustomInvoice.objects.filter(user=user)
+
+    for invoice in custominvoices:
+        invoice.delete()
+
+    subscriptions = ServiceSubscription.objects.filter(user=user)
+    for subscription in subscriptions:
+        subscription.delete()
+
```

Comment:
What about the banktransaction "sender" information?
```
for transaction in BankTransaction.objects.filter(user=user):
  transaction.sender="anonymized"
  transaction.save()
```

(untested)

Reply:
As discussed, sender in bank data is not personal info and isn't necessarily same as the user was.

Reply:
it is personal info

https://tietosuoja.fi/mika-on-henkilotieto#:~:text=Henkil%C3%B6tietoja%20ovat%20kaikki%20tiedot%2C%20jotka,toiseen%20tietoon%2C%20joka%20mahdollistaa%20tunnistamisen.

it doesn't matter where that info exists in the db it is still personal information.

we have no legal basis on keeping that data.